### ISSUE TITLE
[ld65][bug] Cannot set segment offset to $00 explicitly

### ISSUE THREAD

**Issue by domgetter** [cc65/cc65#918]
When setting the offset of a segment to $00 (or anything that calculates to $00), the linker outputs a "Range error".  Expected behavior is that it behaves as if an offset was not specified.

Example:
```ini
SEGMENTS {
    DATA00: load = DATABANKS, type = ro, offset = $2000 * $00;
    DATA01: load = DATABANKS, type = ro, offset = $2000 * $01;
}
```
If that is changed to:
```ini
SEGMENTS {
    DATA00: load = DATABANKS, type = ro;
    DATA01: load = DATABANKS, type = ro, offset = $2000 * $01;
}
```
ld65 correctly links the code, and produces the expected output file.

If the `$2000 * $00` is changed to `$00`, it produces the same error:
```
ld65: Error: ld_cfg_file.cfg(53): Range error
```

**Comment by groessler** [comment]
Could you post complete cfg file, please?

**Comment by domgetter** [comment]
Here is a minimum example:

program.asm
```
.segment "DATA00"
.segment "CODE"
Main:
  JMP Main
```
config.cfg
```
MEMORY {
    DATABANKS: file = %O, start = $0000, size = $1000, fill = yes, define = yes;
}
SEGMENTS {
    DATA00: load = DATABANKS, offset = $0000, type = ro;
    CODE: load = DATABANKS, offset = $0400, type = ro;
}
```
build.bat
```
ca65 program.asm -o program.o
ld65 program.o -C config.cfg -o program.bin
```


And if config.cfg is changed to this:
```
MEMORY {
    DATABANKS: file = %O, start = $0000, size = $1000, fill = yes, define = yes;
}
SEGMENTS {
    DATA00: load = DATABANKS, type = ro;
    CODE: load = DATABANKS, offset = $0400, type = ro;
}
```

then it links without error.

**Comment by greg-king5** [comment]
Fixed by commit 28584b31f1b3560173c16487f45aad23dc6f9fb5.

**Comment by oliverschmidt** [comment]
Thanks :-)
